subsys/net/lib/dns/resolve.c

```diff
@@ -1027,7 +1027,7 @@ static void query_timeout(struct k_work *work)
 	if (ret != 0) {
 		struct k_work_delayable *dwork = k_work_delayable_from_work(work);
 
-		k_work_reschedule(dwork, K_NO_WAIT);
+		k_work_reschedule(dwork, K_MSEC(10));
```

Comment:
Some comment in the code would be good to have, like why we are doing it like this etc. Also good to mention that the timeout value is also arbitrary and can be changed if needed.

Reply:
Updated, please check if it looks good now.